graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -137,33 +143,63 @@ public String getFingerprint() {
      * @return the list of matching streams
      */
     public List<Stream> match(Message message) {
```

Comment:
Tried that. Not possible right now, as it would involve external changes of the API.